test/parallel/begin/deitz/test_begin.chpl

```diff
@@ -3,12 +3,15 @@ use Time;
 const n = 5;
 
 var a: [1..n] int;
+var B: [1..n] sync bool;
 
 for i in 1..n do begin {
   a[i] = foo(i);
+  B[i] = true;
 }
 
 for i in 1..n {
+  B[i];
   while a[i] == 0 do
```

Comment:
With the sync var there to control synchronization you should be able to get rid of the while-loop wait. You should be guaranteed that a(i)!=0 as soon as B(i) is non-empty (as long as foo(i) always returns non-0, of course).